torch/_inductor/kernel/conv.py

```diff
@@ -308,8 +308,20 @@ def convolution(
     dilation = pad_listlike(dilation, ndim)
     output_padding = pad_listlike(output_padding, ndim)
 
+    def channels_last_conv():
```

Comment:
Any intuitive understanding for why this prefers channels last?

Reply:
Since we're permuting the input to `img.permute(0, 2, 3, 1)`, if the input is channels-last, it produces a dense tensor. If the input is contiguous, this produces a bad format for mm.